purchase_sale_inter_company/wizard/stock_return_picking.py

```diff
@@ -0,0 +1,92 @@
+import logging
+
+from odoo import SUPERUSER_ID, _, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class ReturnPicking(models.TransientModel):
+    _inherit = "stock.return.picking"
+
+    def create_returns(self):
+        res = super(ReturnPicking, self).create_returns()
+        pick = self.picking_id
+
+        # dont trigger on returns of incoming pickings
+        if pick.picking_type_code == "incoming":
+            return res
+
+        # only trigger in case there is a coupled picking on the other side
+        ic_pick = pick.intercompany_picking_id
+        if not ic_pick:
+            return res
```

Comment:
We have been working for a couple of weeks with a customer using this PR and it seems like the return of a return make sense in some cases. Is there any reason why this shouldn't be applied?
```suggestion
        # dont trigger on returns of incoming pickings
        if pick.picking_type_code == "incoming" and not pick.intercompany_return_picking_id:
            return res

        # only trigger in case there is a coupled picking on the other side
        ic_pick = pick.intercompany_picking_id or pick.intercompany_return_picking_id
        if not ic_pick:
            return res
```